tests/integration/ha_tests/application_charm/src/charm.py

```diff
@@ -72,14 +72,21 @@ def app_peer_data(self) -> Dict:
     @property
     def _database_config(self):
         """Returns the database config to use to connect to the MongoDB cluster."""
-        data = list(self.database.fetch_relation_data().values())[0]
+        # In some tests we want to write directly to mongos, but the config-server does not
+        # support integrations to client applications, so the data to connect is set via config.
+        data = list(self.database.fetch_relation_data().values())
+        if len(data) == 0:
```

Comment:
nit - good candidate for the walrus operator:
```suggestion
        if not (data := list(self.database.fetch_relation_data().values())):
```